lib/p5.play.js

```diff
@@ -4743,24 +4743,45 @@ function SpriteSheet(pInst) {
 
   if(spriteSheetArgs[0] instanceof p5.Image || spriteSheetArgs[0] instanceof Image) {
     this.image = spriteSheetArgs[0];
-    if (spriteSheetArgs.length === 4) {
+    if (longArgs) {
       this._generateSheetFrames();
     }
   } else {
     // When the final argument is present (either the 3rd or the 5th), it indicates
     // whether we should load the URL as an Image element (as opposed to the default
-    // behavior, which is to load it as a p5.Image)
+    // behavior, which is to load it as a p5.Image). If that argument is a function,
+    // it will be called back once the load succeeds or fails. On success, the Image
+    // will be supplied as the only parameter. On failure, null will be supplied.
+    var callback;
     if (shortArgs) {
       if (spriteSheetArgs[2]) {
-        this.image = pInst.loadImageElement(spriteSheetArgs[0]);
+        if (typeof spriteSheetArgs[2] === 'function') {
+          callback = spriteSheetArgs[2];
```

Comment:
This feels a little gross to me.  Why do we need to overload the "image/p5Image" bit with a callback?  What if I want to create a p5Image but also have a callback?  I'd make the callback a separate argument.

Reply:
yes, totally agree. Unfortunately, we really can't support that many permutations while preserving argument compatibility without making this even worse.
I think we'll need to move to loading the image first (async) and then loading the spritesheet by passing an `Image` (or `p5.Image`) as the first argument (instead of a `url`). When you pass the `url`, you're asking the spritesheet to have to do the loading and there are too many permutations to handle.